url8be3f02398_vnamboo_LearnJava_tgz-pJ8-org_json_TestJ8/src/threads/currentcurrent/Buffer.java

```diff
@@ -0,0 +1,39 @@
+package threads.currentcurrent;
+
+public class Buffer {
+
+	private int data;
+	private boolean empty;
+
+	public Buffer() {
+		this.empty = true;
+	}
+
+	public synchronized void produce(int dataItem) {
+		while (!this.empty) {
+			try {
+				this.wait();
+			} catch (InterruptedException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)